nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/GetSolr.java

```diff
@@ -172,157 +203,196 @@ protected void init(final ProcessorInitializationContext context) {
 
     @Override
     public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, String newValue) {
-        lastEndDatedRef.set(UNINITIALIZED_LAST_END_DATE_VALUE);
+        clearState.set(true);
     }
 
-    @OnStopped
-    public void onStopped() {
-        writeLastEndDate();
-    }
+    @OnScheduled
+    public void onScheduled2(final ProcessContext context) throws IOException {
```

Comment:
Please change method name appropriately to represent what it does, such as `clearState`. The annotation explains when it's called.